failure adding any type of credential

Seeing this at the moment when deployed from scratch from `master`. Happens when adding any type of credentials.

```{"type": "server", "timestamp": "2021-02-11T14:03:43,509Z", "level": "DEBUG", "component": "o.e.a.s.TransportSearchAction", "cluster.name": "elasticsearch", "node.name": "es", "message": "[nuvla-credential][0], node[QvbxMSvXQgy_Xz3GgkqF_Q], [P], s[STARTED], a[id=G5UdY14RS1WfVkzOjRqgzQ]: Failed to execute [SearchRequest{searchType=QUERY_THEN_FETCH, indices=[nuvla-credential], indicesOptions=IndicesOptions[ignore_unavailable=false, allow_no_indices=true, expand_wildcards_open=true, expand_wildcards_closed=false, expand_wildcards_hidden=false, allow_aliases_to_multiple_indices=true, forbid_closed_indices=true, ignore_aliases=false, ignore_throttled=true], types=[], routing='null', preference='null', requestCache=null, scroll=null, maxConcurrentShardRequests=0, batchedReduceSize=512, preFilterShardSize=null, allowPartialSearchResults=true, localClusterAlias=null, getOrCreateAbsoluteStartMillis=-1, ccsMinimizeRoundtrips=true, source={\"from\":0,\"size\":10000,\"query\":{\"bool\":{\"filter\":[{\"bool\":{\"should\":[{\"term\":{\"acl.view-meta\":{\"value\":\"session/9d3d836f-48f2-42aa-b69a-dc31775631f0\",\"boost\":1.0}}},{\"term\":{\"acl.view-meta\":{\"value\":\"group/nuvla-anon\",\"boost\":1.0}}},{\"term\":{\"acl.view-meta\":{\"value\":\"user/c11eaca4-0677-4f7e-8cad-bf320796510e\",\"boost\":1.0}}},{\"term\":{\"acl.view-meta\":{\"value\":\"group/nuvla-user\",\"boost\":1.0}}},{\"term\":{\"acl.owners\":{\"value\":\"session/9d3d836f-48f2-42aa-b69a-dc31775631f0\",\"boost\":1.0}}},{\"term\":{\"acl.owners\":{\"value\":\"group/nuvla-anon\",\"boost\":1.0}}},{\"term\":{\"acl.owners\":{\"value\":\"user/c11eaca4-0677-4f7e-8cad-bf320796510e\",\"boost\":1.0}}},{\"term\":{\"acl.owners\":{\"value\":\"group/nuvla-user\",\"boost\":1.0}}}],\"adjust_pure_negative\":true,\"boost\":1.0}},{\"match_all\":{\"boost\":1.0}}],\"adjust_pure_negative\":true,\"boost\":1.0}},\"sort\":[{\"name\":{\"order\":\"asc\"}},{\"id\":{\"order\":\"asc\"}}]}}]", "cluster.uuid": "DicKQTEBSX2PwtHx7ab0Rw", "node.id": "QvbxMSvXQgy_Xz3GgkqF_Q" ,
"stacktrace": ["org.elasticsearch.transport.RemoteTransportException: [es][10.0.11.9:9300][indices:data/read/search[phase/query]]",
"Caused by: java.lang.IllegalArgumentException: Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [id] in order to load field data by uninverting the inverted index. Note that this can use significant memory.",
"at org.elasticsearch.index.mapper.TextFieldMapper$TextFieldType.fielddataBuilder(TextFieldMapper.java:761) ~[elasticsearch-7.7.0.jar:7.7.0]",
"at org.elasticsearch.index.fielddata.IndexFieldDataService.getForField(IndexFieldDataService.java:116) ~[elasticsearch-7.7.0.jar:7.7.0]",
"at org.elasticsearch.index.query.QueryShardContext.getForField(QueryShardContext.java:221) ~[elasticsearch-7.7.0.jar:7.7.0]",
"at org.elasticsearch.search.sort.FieldSortBuilder.build(FieldSortBuilder.java:403) ~[elasticsearch-7.7.0.jar:7.7.0]",
"at org.elasticsearch.search.sort.SortBuilder.buildSort(SortBuilder.java:159) ~[elasticsearch-7.7.0.jar:7.7.0]",
"at org.elasticsearch.search.SearchService.parseSource(SearchService.java:858) ~[elasticsearch-7.7.0.jar:7.7.0]",
"at org.elasticsearch.search.SearchService.createContext(SearchService.java:678) ~[elasticsearch-7.7.0.jar:7.7.0]",
"at org.elasticsearch.search.SearchService.createAndPutContext(SearchService.java:631) ~[elasticsearch-7.7.0.jar:7.7.0]",
"at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:428) ~[elasticsearch-7.7.0.jar:7.7.0]",
"at org.elasticsearch.search.SearchService.access$200(SearchService.java:135) ~[elasticsearch-7.7.0.jar:7.7.0]",
"at org.elasticsearch.search.SearchService$2.lambda$onResponse$0(SearchService.java:395) ~[elasticsearch-7.7.0.jar:7.7.0]",
"at org.elasticsearch.search.SearchService.lambda$runAsync$0(SearchService.java:411) [elasticsearch-7.7.0.jar:7.7.0]",
"at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44) [elasticsearch-7.7.0.jar:7.7.0]",
"at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:692) [elasticsearch-7.7.0.jar:7.7.0]",
"at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [elasticsearch-7.7.0.jar:7.7.0]",
"at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]",
"at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]",
"at java.lang.Thread.run(Thread.java:832) [?:?]"] }
```

Server-side log.

```
2021-02-11 14:19:43,313 DEBUG - PUT /api/credential [ - ] ?
2021-02-11 14:19:43,339 INFO  - 200 (26 ms) PUT /api/credential [ - ] ?
2021-02-11 14:19:43,390 DEBUG - POST /api/credential [ - ] ?
2021-02-11 14:19:43,448 INFO  - 201 (58 ms) POST /api/credential [ - ] ?
2021-02-11 14:19:43,497 DEBUG - GET /api/credential/bd455023-70f0-4b9c-b839-f7a0cb4ac31a [ - ] ?
2021-02-11 14:19:43,505 DEBUG - PUT /api/credential [ - ] ?
2021-02-11 14:19:43,519 INFO  - 200 (22 ms) GET /api/credential/bd455023-70f0-4b9c-b839-f7a0cb4ac31a [ - ] ?
2021-02-11 14:19:43,523 ERROR - 500 (18 ms) PUT /api/credential [ - ] ?
```

UI always reports OK.

![Screenshot 2021-02-11 at 15 20 46](https://user-images.githubusercontent.com/353767/107648528-b9918b80-6c7c-11eb-9232-49e940ecae2d.png)

The cred can not be found on the server.

--- a/code/src/sixsq/nuvla/server/resources/credential.clj
+++ b/code/src/sixsq/nuvla/server/resources/credential.clj
@@ -45,7 +45,7 @@ passwords) or other services (e.g. TLS credentials for Docker). Creating new
 
 (defn initialize
   []
-  (std-crud/initialize resource-type nil)
+  (std-crud/initialize resource-type ::credential/schema)
   (md/register resource-metadata))
 
 

--- a/code/src/sixsq/nuvla/server/resources/user.clj
+++ b/code/src/sixsq/nuvla/server/resources/user.clj
@@ -208,6 +208,7 @@ requires a template. All the SCRUD actions follow the standard CIMI patterns.
 (defn create-super-user
   [password]
   ;; FIXME: nasty hack to ensure username-password user-template, user-identifier and group index are available
+  (credential/initialize)
   (group/initialize)
   (username-password/initialize)
   (user-identifier/initialize)

--- a/code/test/sixsq/nuvla/server/resources/credential_infrastructure_service_azure_lifecycle_test.clj
+++ b/code/test/sixsq/nuvla/server/resources/credential_infrastructure_service_azure_lifecycle_test.clj
@@ -53,6 +53,13 @@
                                              :azure-client-secret   "def"
                                              :azure-client-id       "ghi"}}]
 
+    ;; check we can perform search with ordering by name
+    (-> session-admin
+        (request (str base-uri "?orderby=name:asc")
+                 :request-method :put)
+        (ltu/body->edn)
+        (ltu/is-status 200))
+
     ;; admin/user query should succeed but be empty (no credentials created yet)
     (doseq [session [session-admin session-user]]
       (-> session

--- a/code/test/sixsq/nuvla/server/resources/credential_ssh_key_lifecycle_test.clj
+++ b/code/test/sixsq/nuvla/server/resources/credential_ssh_key_lifecycle_test.clj
@@ -65,6 +65,13 @@
                           :tags        tags-attr
                           :template    {:href  href}}]
 
+    ;; check we can perform search with ordering by name
+    (-> session-admin
+        (request (str base-uri "?orderby=name:asc")
+                 :request-method :put)
+        (ltu/body->edn)
+        (ltu/is-status 200))
+
     ;; admin/user query should succeed but be empty (no credentials created yet)
     (doseq [session [session-admin session-user]]
       (-> session

